### DISCUSSION TITLE
Can I have function name autocompleted with parentheses?

### DISCUSSION THREAD

**Discussion by zhaob1n** [clangd/clangd#1196] [Q&A] [answered]
I am using an editor called helix. When I use clangd suggestions, the function name is completed without parenthesis.

**Comment by HighCommander4** [comment]
Clangd should include parentheses when completing a function call.

Could you attach clangd logs from an editing session where you invoke completion on a function?

The way to get the logs depends on the editor, this may be a starting point: https://github.com/helix-editor/helix/wiki/FAQ#access-the-log-file

**Comment by sam-mccall** [comment, marked as answer]
We only return the parens if the client claims to support snippets (ClientCapabilities.textDocument.completion.completionItem.snippetSupport) as opposed to plain-text completions.
This is required in order to place the cursor inside the parens by providing the completion snippet `foo($0)`. Without this, completing the parens does more harm than good.

My guess is helix doesn't support snippets.
A recent forum post claiming that: https://forums.swift.org/t/additions-to-codecompletion/56649
